pkg/controller/service/reconciler.go

```diff
@@ -188,6 +188,34 @@ func newDestroyer(invFactory inventory.ClientFactory, f util.Factory) (*apply.De
 		Build()
 }
 
+func enforceNamespace(mans []*unstructured.Unstructured, svc *console.GetServiceDeploymentForAgent_ServiceDeployment) error {
+	if svc == nil {
+		return nil
+	}
+	if svc.SyncConfig == nil {
+		return nil
+	}
+	if svc.SyncConfig.EnforceNamespace == nil {
+		return nil
+	}
+	if !*svc.SyncConfig.EnforceNamespace {
+		return nil
+	}
+
+	for _, man := range mans {
+		gvk := schema.GroupVersionKind{
+			Group:   man.GroupVersionKind().Group,
+			Version: man.GroupVersionKind().Version,
+			Kind:    man.GetKind(),
+		}
+		if controller.ClusterResources.Has(gvk) {
```

Comment:
what about cluster scoped CRD resources not in the fixed list?

Reply:
I have to add them, thanks

Reply:
Yeah I feel like the best approach here is to use `RESTMapper` which should also give you the ability to determine if the GVK is cluster scoped (or something similar in the k8s discovery client, can't fully remember).

Reply:
yes, I found the way how to do this, PTAL